templates/rhsm-katello-reconfigure.erb

```diff
@@ -106,13 +106,19 @@ fi
   service goferd status >/dev/null && \
   service goferd restart >/dev/null 2&>1
 
+[ -f /etc/init.d/goferd ] && \
 [ -f /bin/systemctl ] && \
   systemctl try-restart goferd >/dev/null 2>&1
 
-
-FQDN=`hostname -f`
-if [ $? == "0" ] && [ "$FQDN" != "localhost" ] && [ -d /etc/rhsm/facts/ ]; then
-  echo "{\"network.hostname-override\":\"$FQDN\"}" > /etc/rhsm/facts/katello.facts
+# If RHSM version is less than 1.18.1-1, we have to update the hostname-override fact, since those versions don't
+# have the network.fqdn fact. This doesn't apply to EL5 systems, where we always want to add the hostname-override
```

Comment:
The ```This doesn't apply to EL5``` is slightly misleading. I'd rephrase it as ```EL5 systems and RHSM before 1.18.1-1 are missing the network.fqdn fact so need a hostname-override fact``` or something along those lines.